### PR TITLE
Add support for time-0.2 types

### PR DESCRIPTION
**Pull request by aloucks**
This is essentially ported from the `chrono` implementation.

### CODE REVIEW

File: postgres-types/src/lib.rs [hunk 1]
```diff
@@ -391,6 +397,10 @@ impl WrongType {
 /// | `chrono::DateTime<FixedOffset>` | TIMESTAMP WITH TIME ZONE            |
 /// | `chrono::NaiveDate`             | DATE                                |
 /// | `chrono::NaiveTime`             | TIME                                |
+/// | `time::PrimitiveDateTime`       | TIMESTAMP                           |
+/// | `time::OffsetDateTime`          | TIMESTAMP, TIMESTAMP WITH TIME ZONE |
```
Comment (sfackler) [inline postgres-types/src/lib.rs]: I think this should only support TIMESTAMP WITH TIME ZONE to align with chrono.

**Comment by sfackler** [comment]
Can you add tests as well?

File: postgres-types/src/lib.rs (hunk 1 above)
Comment (aloucks) [reply postgres-types/src/lib.rs]: Chrono's types embed specific timezones (or rather `UTC` vs `Local`) where as `OffsetDateTime` doesn't have the generic component and is compatible with both `TIMESTAMP` and `TIMESTAMPZ`. I'd prefer to be able to use `OffsetDateTime` with `TIMESTAMP`.

**Comment by aloucks** [comment]
> Can you add tests as well?

Ah! I just found the tests under `tokio-postgres`. I'll update and push a commit shortly.

File: postgres-types/src/lib.rs (hunk 1 above)
Comment (aloucks) [reply postgres-types/src/lib.rs]: > I'd prefer to be able to use OffsetDateTime with TIMESTAMP.

With [cockroach](https://www.cockroachlabs.com/docs/v19.2/timestamp.html#variants) `TIMESTAMP` will always be `UTC`. I'm still looking for clarification on postgres's behavior. I think with the fact that we're explicitly setting the session to `UTC`, it should be OK either way.

Comment (aloucks) [reply postgres-types/src/lib.rs]: @sfackler I thought about this some more and came to the same conclusion. `OffsetDateTime` now only supports `TIMESTAMPTZ` in the latest commit. Do you want me to squash these commits?

**Comment by sfackler** [comment]
No need to squash!

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**